src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,103 @@
-/* eslint-disable max-len */
-import React from 'react';
+import classNames from 'classnames';
+import { useCallback } from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { Todo } from '../../types/Todo';
+
+export const TodoList = () => {
+  const todos = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const filter = useAppSelector(state => state.filter);
+  const dispatch = useAppDispatch();
+
+  const handleSelectTodo = (todo: Todo) => {
+    dispatch({ type: 'currentTodo/SET', payload: todo });
+  };
+
+  const filterTodos = useCallback(
+    () => {
+      const cleanQuery = filter?.query.trim().toLowerCase() || '';
+
+      switch (filter?.status) {
+        case 'active':
+          return (todos)
+            .filter(todo => !todo.completed
+                && todo.title.toLowerCase().includes(cleanQuery));
+        case 'completed':
+          return (todos)
+            .filter(todo => todo.completed
+                && todo.title.toLowerCase().includes(cleanQuery));
+        default:
+          return (todos)
+            .filter(todo => todo.title.toLowerCase().includes(cleanQuery));
+      }
```

Comment:
could be quite shorter:
please check whether this would work @UmizDemud 
```suggestion
      return todo.filter(({ title, completed }) => (
        title.toLowerCase().includes(cleanQuery)
        && switch (filter?.status) {
          case 'active':
            return !completed;
          case 'completed':
            return completed;
          default:
            return true;
        }
      ))
```